### PR TITLE
Clean up orphaned participants on chat deletion

### LINKED ISSUE #54: Implement proper cascade deletion in applyChatDeletedDelta
## Problem
Currently `applyChatDeletedDelta()` only deletes the chat entity but doesn't handle cleanup of:
- Associated messages in the chat
- Chat-participant relationships in junction table
- Potentially orphaned participants

This could lead to data inconsistency and storage waste.

## Proposed Solution
Implement proper cascade deletion that:
1. Deletes all messages associated with the chat
2. Removes chat-participant relationships from junction table
3. Considers cleanup of orphaned participants (if not in other chats)
4. Handles foreign key constraints properly

## Impact
- Maintains database consistency
- Prevents data leaks and storage bloat
- Ensures proper cleanup on chat deletion

## Files Affected
- `LocalSyncDataSourceImpl.kt:178-181`
- Related DAO methods
- Potentially Room entity relationships

## Priority
High - Data integrity issue

### PR DESCRIPTION
**Pull request by timurgilfanov**
Closes #54

## Summary
Implements cascade cleanup for deleted chats so local storage does not retain chat-scoped rows or participants that are no longer referenced.

Participant cleanup is conservative: a participant is deleted only when no chat or message still references it. This keeps deleted-chat cleanup from removing participants that remain valid through another chat or retained message history.

## Changes
- Added DAO support for deleting participants that no longer have chat or message references.
- Runs orphan participant cleanup after local chat deletion paths finish removing chat-specific data.
- Expanded DAO and local data source tests for referenced, orphaned, and message-retained participants.
- Replaced non-deterministic test UUID generation with fixed IDs for reproducibility.

## Testing
Validation covered formatting, static analysis, and focused local database unit tests.
- [x] Unit tests added or updated
- [ ] Manually tested
- [x] Edge cases considered

## Screenshots / Demo
Not applicable. This PR does not affect UI.

## Acceptance criteria
- [x] Messages associated with a deleted chat are removed.
- [x] Chat-participant relationships for a deleted chat are removed.
- [x] Participants that are no longer referenced are cleaned up.
- [x] Participants still referenced by another chat or by messages are preserved.
- [x] Foreign key constraints remain respected.

## Breaking Changes
None.

## Follow-ups
None.

### CODE REVIEW

File: app/src/test/java/timur/gilfanov/messenger/data/source/local/LocalSyncDataSourceImplTest.kt [hunk 1]
```diff
@@ -227,6 +228,151 @@ class LocalSyncDataSourceImplTest {
         assertNull(storedChat)
     }
 
+    @Test
+    fun `applyChatDelta with ChatDeletedDelta cascades messages and chat_participants`() = runTest {
```
Comment (timurgilfanov) [inline app/src/test/java/timur/gilfanov/messenger/data/source/local/LocalSyncDataSourceImplTest.kt:232]: Participant have FK to chat, so cascade relate to participant not to cross-reference table.
```suggestion
    fun `applyChatDelta with ChatDeletedDelta cascades messages and participants`() = runTest {
```

Comment (timurgilfanov) [reply app/src/test/java/timur/gilfanov/messenger/data/source/local/LocalSyncDataSourceImplTest.kt:232]: I checked the current schema and this wording looks intentional: `ParticipantEntity` has no FK to `chats`; the FK cascade on chat deletion is `ChatParticipantCrossRef.chatId -> ChatEntity.id`, while `MessageEntity.chatId -> ChatEntity.id` handles messages. Participant rows are removed separately by `deleteOrphanedParticipants()` after the chat deletion, not by the chat FK cascade itself.

I agree the test name can be clearer. A more precise wording would be:

```kotlin
fun `applyChatDelta with ChatDeletedDelta cascades messages and chat participant links`() = runTest {
```
